src/plugins/saved_objects_management/public/management_section/objects_table/components/import_summary.tsx

```diff
@@ -0,0 +1,233 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import './import_summary.scss';
+import React, { Fragment } from 'react';
+import {
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+  EuiIcon,
+  EuiIconTip,
+  EuiHorizontalRule,
+  EuiTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { SavedObjectsImportSuccess } from 'kibana/public';
+import { FailedImport } from '../../..';
+import { getDefaultTitle, getSavedObjectLabel } from '../../../lib';
+
+const DEFAULT_ICON = 'apps';
+
+export interface ImportSummaryProps {
+  failedImports: FailedImport[];
+  successfulImports: SavedObjectsImportSuccess[];
+}
+
+interface ImportItem {
+  type: string;
+  id: string;
+  title: string;
+  icon: string;
+  outcome: 'created' | 'overwritten' | 'error';
+  errorMessage?: string;
+}
+
+const getErrorMessage = ({ error }: FailedImport) => {
+  if (error.type === 'unknown') {
+    return error.message;
+  } else if (error.type === 'unsupported_type') {
+    return i18n.translate(
+      'savedObjectsManagement.objectsTable.importSummary.unsupportedTypeError',
+      { defaultMessage: 'Unsupported object type' }
+    );
+  }
+};
+
+const mapFailedImport = (failure: FailedImport): ImportItem => {
+  const { obj } = failure;
+  const { type, id, meta } = obj;
+  const title = meta.title || getDefaultTitle(obj);
+  const icon = meta.icon || DEFAULT_ICON;
+  const errorMessage = getErrorMessage(failure);
+  return { type, id, title, icon, outcome: 'error', errorMessage };
+};
+
+const mapImportSuccess = (obj: SavedObjectsImportSuccess): ImportItem => {
+  const { type, id, meta, overwrite } = obj;
+  const title = meta.title || getDefaultTitle(obj);
+  const icon = meta.icon || DEFAULT_ICON;
+  const outcome = overwrite ? 'overwritten' : 'created';
+  return { type, id, title, icon, outcome };
+};
+
+const getCountIndicators = (importItems: ImportItem[]) => {
+  if (!importItems.length) {
+    return null;
+  }
+
+  const outcomeCounts = importItems.reduce(
+    (acc, { outcome }) => acc.set(outcome, (acc.get(outcome) ?? 0) + 1),
+    new Map<ImportItem['outcome'], number>()
+  );
+  const createdCount = outcomeCounts.get('created');
+  const overwrittenCount = outcomeCounts.get('overwritten');
+  const errorCount = outcomeCounts.get('error');
+
+  return (
+    <EuiFlexGroup>
+      {createdCount ? (
```

Comment:
Yep it works, I'll make the change.

It's worth noting that `createdCount` will only ever be `undefined` or >= `1`. I'm not sure if this would still work if it could be `0`.